prismicio/slice-machine
# feat: add field dropdown

**Resolves**: DT-2224

### Description

- Rework to have a clean way to manage fields in SM, we will have the two ways to work with fields until we totally finish migrating fields to new design / typescript.
- The goal is now to use `domain/fields.ts` file instead of the old one. The old widgets should start to get less and less use until totally removed.
- Create a new functional component `AddFieldDropdown` with no link at all with old code (that was the goal here).
- Let template management for next PR.
- Align old icons with new icons from new design.
- Update `editor-ui` to have new `DropdownMenu` features.
- Reorder widgets to ensure it's displayed in the order we would like now.

## TODO

- Confirm and validate the copy for all fields
- Update the tooltip preview images

### Checklist





### Preview

https://github.com/user-attachments/assets/b539cfe6-6482-4df9-9d24-117cdf1e31f0

### How to QA [^1]

- Install `slice-machine-ui` alpha `2.2.2-alpha.xru-add-field-dropdown.1`
- Anywhere you would like to add a filed you should not see the modal anymore but the add field dropdown.

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.
